Hyperspace/Classes/Service/Network/NetworkService.swift

```diff
@@ -91,3 +68,59 @@ extension NetworkService: NetworkServiceProtocol {
         tasks.forEach { cancelTask(for: $0.key) }
     }
 }
+
+/// A small helper struct which deals with the conversion of Data, URLResponse and Error objects to NetworkService successes and failures.
+public struct NetworkServiceHelper {
+    
+    /// Used to convert a client error, such as those returned by a NetworkSessionDataTask, into a NetworkServiceResult.
+    ///
+    /// - Parameter clientError: The error returned by the NetworkSessionDataTask.
+    /// - Returns: The outcome NetworkServiceResult dictated by the error.
+    public static func networkServiceFailure(for clientError: Error?) -> NetworkServiceFailure {
```

Comment:
For consistency with `networkServiceResult(for response:)` below, would we want to return a `NetworkServiceResult` here too?

Reply:
We could, but on the use side I ended up then having to immediately unwrap the NetworkServiceResult into the underlying NetworkServiceFailure to use it so I felt like it was a waste in this case to wrap into a Result to simply unwrap immediately after.

Maybe as an addition we could add a third function to the helper that returns the Result?